jsre/jsre.go

```diff
@@ -70,6 +73,28 @@ func New(assetPath string) *JSRE {
 	return re
 }
 
+// randomSource returns a pseudo random float64 in the range [0.0-1.0).
+func randomSource() float64 {
+	var src rand.Source
+	var once sync.Once
+
+	once.Do(func() {
+		bytes := make([]byte, 8)
+		seed := time.Now().UnixNano()
+		if _, err := crand.Read(bytes); err == nil {
+			seed = int64(binary.LittleEndian.Uint64(bytes))
+		}
+		src = rand.NewSource(seed)
+	})
```

Comment:
Also I'm not sure whether otto allows this random source to be used concurrently. If so, a lock is needed within the anonymous function. This should probably be checked inside the otto vm code, but to be on the safe side, I'd add it.